core/templates/dev/head/pages/topic_editor/TopicEditor.js

```diff
@@ -21,8 +21,14 @@ oppia.constant(
 oppia.constant(
   'TOPIC_MANAGER_RIGHTS_URL_TEMPLATE',
   '/rightshandler/assign_topic_manager/<topic_id>/<assignee_id>');
+oppia.constant(
+  'TOPIC_RIGHTS_URL_TEMPLATE', '/rightshandler/get_topic_rights/<topic_id>');
+
+oppia.constant(
+  'TOPIC_NAME_INPUT_FOCUS_LABEL', 'topicNameInputFocusLabel');
 
-oppia.controller('TopicEditor', [
-  function() {
+oppia.controller('TopicEditor', ['$scope', 'TopicEditorStateService',
+  function($scope, TopicEditorStateService) {
+    TopicEditorStateService.loadTopic(GLOBALS.topicId);
```

Comment:
Try to avoid using GLOBALS if possible. Have a TopicIdService that retrieves the topic ID from the URL. Ditto elsewhere.

/cc @vojtechjelinek @1995YogeshSharma @BenHenning -- can you confirm that the above suggestion is the correct way to handle this?

Reply:
Even in collections, this is done, should I change both places?

Reply:
Yes, I think so. Though would be good to get confirmation from at least one of @vojtechjelinek @1995YogeshSharma @BenHenning first.

Reply:
Yes, if it's possible in this case, then retrieve the topicId from the URL.

Reply:
I actually think we used this with collections as an alternative to parsing the URL like the exploration player does. I found it a bit cleaner since the ID was passed directly from the backend. That being said, I think we also isolated the GLOBALS usage to the controller's constructor and otherwise passed around a collectionId variable where we needed it.

I think I prefer not parsing the URL since it seems a bit safer/more reliable to depend on the exact value passed by the backend rather than parsing out the URL.

Reply:
Ah, @BenHenning -- I think we're working to get rid of GLOBALS across the site. Do you have alternative suggestions for how to handle this if GLOBALS is not available? I'm not sure I see an alternative...

Though, @vojtechjelinek @1995YogeshSharma, this brings up a good point that we should find a way to load the 404 error page if the ID in a URL is invalid and this is discovered on ajax load...

Reply:
Is the idea to get rid of GLOBALS because the global state being shared here is a bit awkward, or because of the broader Jinja work? If the latter, don't we still need a mechanism for posting data to the page outside of the URL?

I don't have a strong opinion here since the argument makes sense for both approaches.

Reply:
The latter. And to the follow-up question, I think we don't, actually. We'll use subsequent AJAX calls to get the necessary data from the backend.

Reply:
(Though please let us know if you have follow-on thoughts to this!)